python/demo/demo_biharmonic.py

```diff
@@ -112,15 +112,14 @@
 
 # +
 import numpy as np
-
 import ufl
-from dolfinx import fem, io, mesh, plot
 from dolfinx.mesh import CellType, GhostMode
+from mpi4py import MPI
+from petsc4py.PETSc import ScalarType
```

Comment:
@garth-wells is there any reason why the new `default_scalar_type` is not used here?

Reply:
Logic was that these demos import `petsc4py` functionality via DOLFINx modules with petsc in the name, so it's not a problem to get the scalar type from `petsc4py`. I made the change in the main part of the code to avoid importing `petsc4py` at lower levels and where `petsc4py` is not used, e.g. in `forms.py`.